### PR TITLE
Remove unnecessary LazyPublishArtifact constructors

### PR DESCRIPTION
**Pull request by tresat**
These constructors weren't in use by the latest version (2.7.8) of Spring Boot, they were only used by the Play plugin.  After building a new release of that plugin, they can be removed.

This also updates many applications of Spring Boot in snippets and samples to 2.7.8 and adjusts `ArtifactFile` to document and demonstrate providing a `null` argument for version.

### CODE REVIEW

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

**Review by hythloda — APPROVED** [review]

**Comment by tresat** [comment]
@bot-gradle test this

File: subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AbstractSmokeTest.groovy [hunk 1]
```diff
@@ -120,7 +120,7 @@ abstract class AbstractSmokeTest extends Specification {
         static gradleVersions = "0.42.0"
 
         // https://plugins.gradle.org/plugin/org.gradle.playframework
-        static playframework = "0.12"
+        static playframework = "0.13"
```
Comment (big-guy) [inline subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AbstractSmokeTest.groovy:123]: Do we also need to update springBoot's version in this file too? It looks like we're testing with 2.7.3, but maybe we're not trying to publish anything?

**Review by big-guy — APPROVED** [review]

File: subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AbstractSmokeTest.groovy (hunk 1 above)
Comment (tresat) [reply subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AbstractSmokeTest.groovy:123]: Updated.  This removed an expected failure due to corrected input annotations.

**Comment by tresat** [comment]
@bot-gradle test and merge

**Comment by tresat** [comment]
@bot-gradle test and merge

**Pull request merged**